services/continuous_querier/service_test.go

```diff
@@ -61,6 +62,45 @@ func TestExecuteContinuousQuery_HappyPath(t *testing.T) {
 	}
 }
 
+// Test ExecuteContinuousQuery when INTO measurements are taken from the FROM clause.
+func TestExecuteContinuousQuery_ReferenceSource(t *testing.T) {
+	s := NewTestService(t)
+	dbis, _ := s.MetaStore.Databases()
+	dbi := dbis[2]
+	cqi := dbi.ContinuousQueries[0]
+
+	rowCnt := 2
+	pointCnt := 1
+	qe := s.QueryExecutor.(*QueryExecutor)
+	qe.Results = []*influxql.Result{genResult(rowCnt, pointCnt)}
+
+	pw := s.PointsWriter.(*PointsWriter)
+	pw.WritePointsFn = func(p *cluster.WritePointsRequest) error {
+		if len(p.Points) != pointCnt*rowCnt {
+			return fmt.Errorf("exp = %d, got = %d", pointCnt, len(p.Points))
+		}
+
+		exp := "cpu,host=server01 value=0"
+		got := p.Points[0].String()
+		if !strings.Contains(got, exp) {
+			return fmt.Errorf("\n\tExpected ':MEASUREMENT' to be expanded to the measurement name(s) in the FROM regexp.\n\tqry = %s\n\texp = %s\n\tgot = %s\n", cqi.Query, got, exp)
+		}
+
+		exp = "cpu2,host=server01 value=0"
+		got = p.Points[1].String()
+		if !strings.Contains(got, exp) {
+			return fmt.Errorf("\n\tExpected ':MEASUREMENT' to be expanded to the measurement name(s) in the FROM regexp.\n\tqry = %s\n\texp = %s\n\tgot = %s\n", cqi.Query, got, exp)
+		}
+
+		return nil
+	}
+
+	err := s.ExecuteContinuousQuery(&dbi, &cqi)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
 // Test the service happy path.
 func TestService_HappyPath(t *testing.T) {
```

Comment:
unrelated to this PR, but why is this test named this? `TestService` should actually be something like `TestContinuousQueryService` and `HappyPath`?